sql/2025/05/featured-image-block-usage.sql

```diff
@@ -0,0 +1,73 @@
+# HTTP Archive query to get % WordPress sites using Featured Image in block themes.
+#
+# WPP Research, Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# See query results here: https://github.com/GoogleChromeLabs/wpp-research/pull/184
+
+DECLARE
+  DATE_TO_QUERY DATE DEFAULT '2025-03-01';
+
+CREATE TEMPORARY FUNCTION IS_CMS(technologies ARRAY<STRUCT<technology STRING, categories ARRAY<STRING>, info ARRAY<STRING>>>, cms STRING, version STRING) RETURNS BOOL AS (
+  EXISTS(
+    SELECT * FROM UNNEST(technologies) AS technology, UNNEST(technology.info) AS info
+    WHERE technology.technology = cms
+    AND (
+      version = ""
+      OR ENDS_WITH(version, ".x") AND (STARTS_WITH(info, RTRIM(version, "x")) OR info = RTRIM(version, ".x"))
+      OR info = version
+    )
+  )
+);
+
+WITH wordpress AS (
+  SELECT
+    client,
+    page
+  FROM
+    `httparchive.crawl.pages`
+  WHERE
+    date = DATE_TO_QUERY
+    AND IS_CMS(technologies, 'WordPress', '5.9.0')
```

Comment:
Note that `5.9.0` would only match exactly this major version. You'd want `5.9.x` to include minor releases as well. However, what we're interested in are all sites on 5.9 _or newer_, not just sites on 5.9 (which would be a very small number). For that we'd need to modify the query slightly to do a version number comparison, or do a few `or is 6.0.x, or is 6.1.x, ...` etc.

But I think we can keep this simple, because most sites would be on newer versions anyway. So let's just ignore the WordPress version number altogether:

```suggestion
    AND IS_CMS(technologies, 'WordPress', '')
```